src/elements/isinstock-button.tsx

```diff
@@ -35,10 +35,11 @@ const IsInStockButton = ({productValidation}: IsInStockButtonProps) => {
       target="_blank"
       class="btn"
       rel="noreferrer"
+      data-inventory-state={productValidation.availability}
       data-inventory-state-normalized={InventoryStateNormalized.Available}
     >
       <img class="isinstock-logo" width="16" height="16" src={availableImg} />
-      <span>In Stock</span>
+      <span>{productValidation.availability === 'PreOrder' ? 'Pre-Order' : 'In Stock'}</span>
```

Comment:
Likely should get the string representation from the server (one place), or some i18n lookup on the client based on the normalized value (`PreOrder: "Pre-Order", OutOfStock: "Notify Me When Available"`, etc)